plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/AbstractMetastoreTableOperations.java

```diff
@@ -113,7 +117,28 @@ protected final void commitNewTable(TableMetadata metadata)
         Table table = builder.build();
 
         PrincipalPrivileges privileges = owner.map(MetastoreUtil::buildInitialPrivilegeSet).orElse(NO_PRIVILEGES);
-        metastore.createTable(table, privileges);
+        try {
+            metastore.createTable(table, privileges);
+            cleanupOnFailure = false;
+        }
+        catch (SchemaNotFoundException
+               | TableAlreadyExistsException
+               | UnsupportedOperationException e) {
+            // Signal a non-retriable commit failure and eventually clean up metadata files corresponding to the current transaction
```

Comment:
typo `retriable` 

In any case there is no retry performed in case of creating tables (compared to the situation of committing to existing tables).

Reply:
removed `a non-retriable` from comment.